src/api/wayfire/view.hpp

```diff
@@ -491,6 +491,11 @@ class view_interface_t : public surface_interface_t, public wf::object_base_t
      * or activated state changes */
     virtual void desktop_state_updated();
 
+    /**
+     * Called when the parent of a toplevel view is changed.
+     * Needed for the xdg_foreign_toplevel interface. */
+    virtual void parent_updated();
```

Comment:
Seems this can be grouped together with `desktop_state_updated()` to avoid having multiple functions (after all parent is kind of setting transient-for state)

Reply:
My intention was to only send the parent event if there is a change in parent (a strict interpretation of the protocol). To avoid adding to the interface, this check could be done in the wlroots code as well.
Also, I've noticed that Wayfire / wlroots sends a bunch of state events even if the state did not change; I might submit a separate PR to wlroots for that.